lib/sql-logging/statistics.rb

```diff
@@ -10,80 +31,87 @@ def configuration
                      :show_top_sql_queries, :backtrace_cleaner,
                      :show_top_sql_queries=, :show_sql_backtrace=,
                      :top_sql_queries=
-    end
 
-    @@queries = @@bytes = @@rows = 0
-    @@top_queries = {}
+      @data = Data.new
 
-    def self.reset_statistics!
-      @@queries = @@bytes = @@rows = 0
-      @@top_queries = {}
-    end
+      def_delegator :@data, :top_queries
+
+      def reset_statistics!
+        @data.reset!
+      end
+
+      def record_query(sql, name, msec, result)
+        return if name.blank? || name =~ / Columns$/ || name == :skip_logging
+        ntuples, bytes = tuples_and_bytes_in_result(result)
+
+        @data.add_query(ntuple, bytes)
```

Comment:
`ntuple` should be `ntuples`. I know you fixed this typo in the following commit, but I'd've preferred to see the fix squashed into this commit before opening the PR.